src/main/java/seedu/binbash/parser/UpdateCommandParser.java

```diff
@@ -0,0 +1,163 @@
+package seedu.binbash.parser;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+
+import seedu.binbash.command.UpdateCommand;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class UpdateCommandParser extends DefaultParser {
+    protected static final DateTimeFormatter EXPECTED_INPUT_DATE_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+
+    public UpdateCommandParser() {
+        options = new Options();
+        new CommandOptionAdder(options)
+                .addItemNameAndIndexOptionGroup()
+                .addDescriptionOption(false, "Update description of item")
+                .addQuantityOption(false, "Update quantity of item")
+                .addCostPriceOption(false, "Update cost price of item")
+                .addSalePriceOption(false, "Update sale price of item")
+                .addExpirationDateOption(false,
+                        "Update expiry date of item")
+                .addThresholdOption(false,
+                        "Update threshold of item");
+    }
+
+    public UpdateCommand parse(String[] commandArgs) throws ParseException {
+        CommandLine commandLine = new DefaultParser().parse(options, commandArgs);
+
+        boolean hasOption = false;
+        UpdateCommand updateCommand;
+
+        updateCommand = getUpdateCommand(commandLine);
+
+        if (commandLine.hasOption("description")) {
+            hasOption = hasDescriptionOption(commandLine, updateCommand);
+        }
+        if (commandLine.hasOption("quantity")) {
+            hasOption = hasQuantityOption(commandLine, updateCommand);
+        }
+        if (commandLine.hasOption("cost-price")) {
+            hasOption = hasCostPriceOption(commandLine, updateCommand);
+        }
+        if (commandLine.hasOption("sale-price")) {
+            hasOption = hasSalePriceOption(commandLine, updateCommand);
+        }
+        if (commandLine.hasOption("expiry-date")) {
+            hasOption = hasExpirationDateOption(commandLine, updateCommand);
+        }
+        if (commandLine.hasOption("threshold")) {
+            hasOption = hasThresholdOption(commandLine, updateCommand);
+        }
+        if (!hasOption) {
+            throw new ParseException("At least one of -n, -d, -c, -s, -e option required");
+        }
+        return updateCommand;
+    }
+
+    private static boolean hasThresholdOption(CommandLine commandLine, UpdateCommand updateCommand)
+            throws ParseException {
+        boolean hasOption;
+        String threshold = commandLine.getOptionValue("threshold");
+        int itemThreshold = Integer.parseInt(threshold);
```

Comment:
Unhandled NumberFormatException. Consider using the Apache commons library TypeHandler class.
```suggestion
        int itemThreshold = TypeHandler.createNumber(commandLine.getOptionValue("threshold")).intValue();
```

Reply:
Got it!